### PR TITLE
Added isReachableStatic and isInCollisionStatic and transformToBaseStatic

### PR DESCRIPTION
**Pull request by geoffreychiou**
The original functions called transformToBase, which made a tf lookup for every single request causing lots of lag. These functions assume the transform to be static and will only look it up for the first instance.

@joshuaplusone please review.

### CODE REVIEW

**Comment by geoffreychiou** [comment]
Setting to WIP as @joshuaplusone has some requested changes.

File: moveit_simple/test/kuka_kr210.cpp [hunk 1]
```diff
@@ -647,15 +647,18 @@ TEST_F(UserRobotTest, copy_current_pose)
   // Before the first execution robot should be at position "home"
   before_execution_1 = robot->getJointState();
   EXPECT_NO_THROW(robot->execute(TRAJECTORY_NAME));
+  ros::Duration(0.5).sleep();
```
Comment (shaun-edwards) [inline moveit_simple/test/kuka_kr210.cpp:650]: Why is this needed?

**Review by shaun-edwards — DISMISSED** [review]

File: moveit_simple/test/kuka_kr210.cpp (hunk 1 above)
Comment (geoffreychiou) [reply moveit_simple/test/kuka_kr210.cpp:650]: I think we had this discussion for the mh5 tests, added something similar [there](https://github.com/plusone-robotics/moveit_simple/blob/kinetic-devel/moveit_simple/test/motoman_mh5.cpp#L750). It was something about the industrial_robot_simlator was still at the end of the execution before the next getJointState() was called, causing the tests to sometimes fail or sometimes pass.

**Comment by geoffreychiou** [comment]
@joshuaplusone refactored as your requested, this is review ready.

File: moveit_simple/include/moveit_simple/robot.h [hunk 2]
```diff
@@ -87,20 +112,21 @@ class Robot
    * @param joint_seed (optional) - seed to use
    * @return
    */
-  bool isInCollision(const Eigen::Affine3d pose, const std::string &frame,
+  bool isInCollision(const Eigen::Affine3d &pose, const std::string &frame,
```
Comment (joshuaplusone) [inline moveit_simple/include/moveit_simple/robot.h:115]: Nice catch.

**Review by joshuaplusone — APPROVED** [review]
Looks good to me.
